thalerj/openbmctool.py

```diff
@@ -0,0 +1,1490 @@
+            "\t\t\"UserAction\": \"Correct the issue highlighted in additional details and try again\"" +
+            "\t\n}, \n" +
+            "\t\"numAlerts\": \"1\" \n}");
+            return(errorMessageStr)
+    else:
+        return("Unknown Error: "+ str(err))
+
+"""
+     Sets the output width of the columns to display
+       
+     @param keylist: list, list of strings representing the keys for the dictForOutput 
+     @param numcols: the total number of columns in the final output
+     @param dictForOutput: dictionary, contains the information to print to the screen
+     @param colNames: list, The strings to use for the column headings, in order of the keylist
+     @return: A list of the column widths for each respective column. 
+"""
+def setColWidth(keylist, numCols, dictForOutput, colNames):
+    colWidths = []
+    for x in range(0, numCols):
+        colWidths.append(0)
+    for key in dictForOutput:
+        for x in range(0, numCols):
+            colWidths[x] = max(colWidths[x], len(str(dictForOutput[key][keylist[x]])))
+    
+    for x in range(0, numCols):
+        colWidths[x] = max(colWidths[x], len(colNames[x])) +2
+    
+    return colWidths
+
+def loadPolicyTable(pathToPolicyTable):
+    policyTable = {}
+    if(os.path.exists(pathToPolicyTable)):
+        with open(pathToPolicyTable, 'r') as stream:
+            try:
+#                 contents = yaml.load(stream)
+                contents =json.load(stream)
+                policyTable = contents['events']
+            except yaml.YAMLError as err:
+                print(err)
+    return policyTable
+
+"""
+     Logs into the BMC and creates a session
+       
+     @param host: string, the hostname or IP address of the bmc to log into
+     @param username: The user name for the bmc to log into
+     @param pw: The password for the BMC to log into
+     @param jsonFormat: boolean, flag that will only allow relevant data from user command to be display. This function becomes silent when set to true. 
+     @return: Session object
+"""
+def login(host, username, pw,jsonFormat):
+    if(jsonFormat==False):
+        print("Attempting login...")
+    httpHeader = {'Content-Type':'application/json'}
+    mysess = requests.session()
+    try:
+        r = mysess.post('https://'+host+'/login', headers=httpHeader, json = {"data": [username, pw]}, verify=False, timeout=30)
+        loginMessage = json.loads(r.text)
+        if (loginMessage['status'] != "ok"):
+            print(loginMessage["data"]["description"].encode('utf-8')) 
+            sys.exit(1)
+#         if(sys.version_info < (3,0)):
+#             urllib3.disable_warnings(urllib3.exceptions.InsecureRequestWarning)
+#         if sys.version_info >= (3,0):
+#             requests.packages.urllib3.disable_warnings(requests.packages.urllib3.exceptions.InsecureRequestWarning)
+        return mysess
+    except(requests.exceptions.Timeout):
+        print(connectionErrHandler(jsonFormat, "Timeout", None))
```

Comment:
Why are we paying a descriptive string for the exception as well as the exception object? This feels redundant: I feel like we can just pass the exception and use `isinstance()` to test its type in `connectionErrHandler()`.